app/graphql/types/system.rb

```diff
@@ -30,6 +31,15 @@ class System < Types::BaseObject
                required: false
     end
 
+    def profiles(lookahead:)
+      if %i[rulesPassed rulesFailed compliant lastScanned].any? do |option|
+        lookahead.selects?(option)
+      end
+        context[:parent_system_id] = object.id
+      end
```

Comment:
What's with this extra `end` ?

Reply:
This is the `end`  of the `if`, the one above is the `do end` block. I'll move it to a separate method for clarity